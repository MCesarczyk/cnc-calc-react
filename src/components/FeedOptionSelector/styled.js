import styled from "styled-components";

export const FormLabel = styled.label`
    display: flex;
    gap: 5px;
    align-items: center;
    justify-content: space-between;

    @media (max-width: 480px) {
        flex-direction: column;
        justify-content: center;
    }
`;

export const RadioButtonContainer = styled.div`
    display: flex;
    gap: 10px;
`;