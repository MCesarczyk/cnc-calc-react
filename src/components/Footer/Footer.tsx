import { AUTHOR_PAGE_ADDRESS, AUTHOR_PAGE_LABEL, DATE } from 'assets/fixtures/author';
import styled from 'styled-components';

export const Footer = () => (
  <FooterWrapper>
    <FooterContent>
      Copyright &copy; {DATE} by&nbsp;
      <FooterLink href={AUTHOR_PAGE_ADDRESS}>{AUTHOR_PAGE_LABEL}</FooterLink>
    </FooterContent>
  </FooterWrapper>
);

const FooterWrapper = styled.footer`
  text-align: center;
  margin: 0;
`;

const FooterContent = styled.p`
  color: ${({ theme }) => theme.color.primary};
  padding: 0 25px;
  font-size: 14px;
`;

const FooterLink = styled.a`
  color: inherit;
  transition: 0.5s ease-out;

  &:hover {
    color: crimson;
  }
`;
