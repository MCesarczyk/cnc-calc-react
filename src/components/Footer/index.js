import { StyledFooter, FooterContent } from "./styled";

const Footer = ({ children }) => (
  <StyledFooter>
    <FooterContent>
      {children}
    </FooterContent>
  </StyledFooter>
);

export default Footer;