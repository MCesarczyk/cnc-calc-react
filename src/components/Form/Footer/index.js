import { useContext } from "react";
import LanguageContext from "../../../features/language/context";
import languages from "../../../assets/fixtures/languages";
import { ButtonsContainer, FooterButton } from "./styled";

const FormFooter = () => {
    const { langId } = useContext(LanguageContext);

    return (
        <ButtonsContainer>
            <FooterButton type="submit">
                {languages[langId]?.buttonLabel.submit}
            </FooterButton>
            <FooterButton type="reset">
                {languages[langId]?.buttonLabel.reset}
            </FooterButton>
        </ButtonsContainer>
    )
};

export default FormFooter;