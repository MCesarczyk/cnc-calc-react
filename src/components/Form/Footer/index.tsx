import { ButtonsContainer, FooterButton } from "./styled";

interface FormFooterProps {
  buttons: {
    id: number;
    label: string;
    type: "submit" | "reset";
  }[];
}

const FormFooter = ({ buttons }: FormFooterProps) => (
  <ButtonsContainer>
    {buttons.map((button) => (
      <FooterButton key={button.id} type={button.type}>
        {button.label}
      </FooterButton>
    ))}
  </ButtonsContainer>
);

export default FormFooter;
