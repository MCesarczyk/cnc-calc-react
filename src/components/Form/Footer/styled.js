import styled from "styled-components";

export const ButtonsContainer = styled.div`
    max-width: 30rem;
    margin: 0 auto;
    display: flex;
    justify-content: center;
    gap: 0.75rem;

    @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
        flex-direction: column;
        align-items: stretch;
    }
`;