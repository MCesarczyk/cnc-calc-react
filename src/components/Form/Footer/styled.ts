import styled, { css } from 'styled-components';

export const ButtonsContainer = styled.div`
  max-width: 30rem;
  margin: 0 auto;
  display: flex;
  justify-content: center;
  gap: 0.75rem;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    flex-direction: column;
    align-items: stretch;
  }
`;

export const FooterButton = styled.button`
  display: flex;
  min-width: 12rem;
  width: 100%;
  justify-content: center;
  color: ${({ theme }) => theme.color.primaryContrast};
  background-color: ${({ theme }) => theme.color.primary};
  border: solid ${({ theme }) => theme.color.primary};
  padding: 0.25rem;
  border-radius: 1.5rem;

  &:hover {
    filter: brightness(150%);
  }

  &:active {
    filter: brightness(200%);
    border-color: currentColor;
  }

  ${({ type }) =>
    type === 'submit' &&
    css`
      color: ${({ theme }) => theme.color.submitButtonFont};
    `}

  ${({ type }) =>
    type === 'reset' &&
    css`
      color: ${({ theme }) => theme.color.resetButtonFont};
    `}
`;
