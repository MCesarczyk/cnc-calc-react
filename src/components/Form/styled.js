import styled from "styled-components";

export const FormWrapper = styled.form`
    color: ${({ theme }) => theme.color.primary};
    margin-bottom: 2rem;
`;

export const Fieldset = styled.fieldset`
    border-radius: 2rem;
    padding: 1rem;
`;

export const Legend = styled.legend`
    text-align: center;
    margin: 0.5rem
`;

export const FieldsContainer = styled.div`
    display: grid;
    justify-content: center;
    gap: 1rem;
    margin-bottom: 1.5rem;

    @media (max-width: 480px) {
        gap: 0.5rem;
    }
`;