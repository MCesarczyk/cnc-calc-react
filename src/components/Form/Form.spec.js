import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";
import "jest-styled-components";
import { FieldsContainer } from "./styled";
import { ButtonsContainer, FooterButton } from "./Footer/styled";

test("Fields container should have smaller gap on smallest mobile screen", () => {
  const containerComponent = renderer.create(
    <ThemeProvider theme={theme}>
      <FieldsContainer />
    </ThemeProvider>
  );

  const container = containerComponent.toJSON();
  expect(container).toHaveStyleRule('gap', '1rem');
  expect(container).toHaveStyleRule('gap', '0.5rem', {
    media: '(max-width:480px)'
  });
});

test("Buttons change direction to column on mobile", () => {
  const buttonsContainerComponent = renderer.create(
    <ThemeProvider theme={theme}>
      <ButtonsContainer />
    </ThemeProvider>
  );

  const buttonsContainer = buttonsContainerComponent.toJSON();
  expect(buttonsContainer).toHaveStyleRule('flex-direction', 'column', {
    media: '(max-width:480px)'
  });

  expect(buttonsContainer).toHaveStyleRule('align-items', 'stretch', {
    media: '(max-width:480px)'
  });
});

test("Form Button with states rendering properly", () => {
  const buttonComponent = renderer.create(
    <ThemeProvider theme={theme} >
      <FooterButton />
    </ThemeProvider>
  );

  let button = buttonComponent.toJSON();
  expect(button).toMatchSnapshot();

  expect(button).toHaveStyleRule('filter', 'brightness(150%)', {
    modifier: ':hover'
  });

  expect(button).toHaveStyleRule('filter', 'brightness(200%)', {
    modifier: ':active'
  });

  expect(button).toHaveStyleRule('border-color', 'currentColor', {
    modifier: ':active'
  });
});