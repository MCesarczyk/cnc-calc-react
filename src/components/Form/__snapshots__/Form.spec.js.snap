// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Button with states rendering properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 12rem;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #fff;
  background-color: #1e3246;
  border: solid #1e3246;
  padding: 0.25rem;
  border-radius: 1.5rem;
}

.c0:hover {
  -webkit-filter: brightness(150%);
  filter: brightness(150%);
}

.c0:active {
  -webkit-filter: brightness(200%);
  filter: brightness(200%);
  border-color: currentColor;
}

<button
  className="c0"
/>
`;
